src/core-asyncpipeline/AsyncPipeline/AsyncPipeline.T/ToTask/ToTask.cs

```diff
@@ -0,0 +1,13 @@
+#nullable enable
+
+using System.Threading.Tasks;
+
+namespace System
+{
+    partial struct AsyncPipeline<T>
+    {
+        public Task<T> ToTask()
+            =>
+            isCanceled is false ? valueTask.AsTask() : Task.FromCanceled<T>(CanceledToken());
```

Comment:
Or we can just rename `ToTask()` to `AsTask()` - without any wrapping.
But in this case the implicit conversion should be removed (because `As` is not a conversion).

Reply:
Decided to keep as is.